### DISCUSSION TITLE
Scrum Prep Milestone 1-1

### DISCUSSION THREAD

**Discussion by NatiPerez459** [uprm-inso4101-2024-2025-s1/semester-project-mercado-colegial-online#27] [General]
Here team members will post a short progress report of what they have been working on during the last week. This should be completed before Friday Sept 6, 11:00AM.

**Comment by DiegoVazquez1** [comment]
Created a issue/task of developing the main user class that will have 2 subclasses of buyer and seller. Created the User class with setters and getters and a display function to see how the log-in/sign-up page is looking. Had to do some research on javascript, html and css to get the live server running and encountered an issue with Node.js that will soon be resolved

**Comment by JuanDelPueblo** [comment]
Created a task for implementing password hashing which will improve security. Helped troubleshoot an issue with the sign up logic code not running with Node.js, which will need further development to make it work properly.
p.s. my name is Edyan Cruz

**Comment by JuanIranzo** [comment]
Created a task for developing an Entity Relation Diagram (ERD) for the design of the databse. Worked on this same task and provided an initial ERD, which was used as a strong working base for the latest proposed ERD. Opened a task for the creation of a shared documentation doc that follows guidelines. Finally, worked on researching the steps necessary to implement a database, provided a summary of the video and the link to the original video. I had a delicate family situation so I could not do as much research as I had liked to (I informed @david-baez-bravo of the situation)

**Comment by jonael-garcia26** [comment]
I completed the task of searching for information on the creation of login/register pages and how the Frontend communicates with the Database. I shared the information I found in a Google doc. I also finished the task of creating empty folders and files that I found necessary for the organization of the project. I have limited knowledge of web development, so I might be wrong, but I think the .json files could cause issues when downloading dependencies, so they might need to be deleted for now (we'll see).

**Comment by PerezMarcelo** [comment]
Created the task to create a basic HTML sign-up form with fields for username, email, and password. The form should send data to the backend code for processing. This way we cn test the backend logic for the sign up and log in. Started working on my assigned task that is to create the Main User class.

**Comment by cristian-barreras** [comment]
I have no experience in web development, so I have been watching the shared resources to familiarize myself with sending and receiving data from the database.

**Comment by YampierFernandez** [comment]
Participated in the task focused on establishing the entity relation diagram for database design. Met with another team lead to discuss ideas for task and future organization. Finally, created three tasks for database design: 1 for researching a proper programing language, 1 for researching a proper code editor, and 1 for investigating and sharing resources for the team to learn.

**Comment by JadStelar** [comment]
Jahdiel Montero
Completed the first iteration of tech stack documentation detailing the technologies used by each team throughout the semester (this will be updated as more technologies get introduced/discontinued). The current iterations of the database schema and website UI design sketch were also added for reference. Additional general information about git and a helpful git commit message template were added for cleaner commit logs.

**Comment by CovasJJ** [comment]
Jose Covas
Started working on the task of searching for a Database Hosting Service that fits our needs. Watched the videos and read the article presented by the team to familiarize myself more with the subject. I documented the information I gathered from my research in the comment section of my task.

**Comment by Kaariinaa08** [comment]
For this week, I focused on researching different web designs and templates specifically for online stores to guide the development of our Product Listing Page layout. I analyzed various e-commerce sites to understand how they structure product grids, filters, and sorting options, aiming to create a user-friendly and visually appealing design for our store. 

- Karina Lopez

**Comment by franrov4** [comment]
Met with fellow team lead to discuss ideas, as well as further scale mapping of the database. Currently working on the setup of our database and hosting service along with our team members. Database team ended up creating various tasks for the first week, hoping to have them done as soon as possible.

-Francisco Rovira

**Comment by CdoxHype** [comment]
Created a task to represent the Schema of the website, how the pages are connected to each other and which ones should be in it. I also helped with the task of the web GUI.

-Julio Camacho

**Comment by RyanVelez101** [comment]
I couldn’t progress on any work this week due to being sick. 

-Ryan Velez

**Comment by elliot-gomez** [comment]
This week, I reached out to the my web development team leads and the project managers to see if there were any immediate steps or tasks I could take to support the team. At the time, we were waiting for the website design to be completed, so I was temporarily put on hold. Later, I was assigned the task of creating the documentation doc with the required guidelines. I found that the task could be more specific and divided into different tasks for all the developers assigned to it, so I brought this up to the project managers. After proposing to divide the task into subtasks, the task was divided into subtasks, one for each developer assigned. I took responsibility for the first subtask I proposed, researching and selecting a suitable documentation format, and began working on it. Additionally, I created the "Document Review and Quality Assurance" task after further discussions with the PM, to ensure the documentation aligns with project standards.

- Elliot J. Gomez Coriano

**Comment by AlvinBadillo** [comment]
This week I was tasked to research and select the best database for our project. I posted the resources I used in my tasks so other developer can also study the selected database. I ended up selecting MongoDb. I am also currently tasked to set up a base Node.js server. This task is currently on-going and will be finished by the stated deadline in the task.

- Alvin Y. Badillo

**Comment by Christian2101** [comment]
I have no experience in database so I started watching videos and researching about the subject. Also started working on the task of searching for a Database Hosting Service. Watched the videos and read the article presented by the team to familiarize myself more with the subject. I documented the information I gathered from my research in the comment section of my task.

**Comment by FabiolaZTorres** [comment]
Researched how to implement the backend code for the sign-up page with the tools that were selected for the tech stack. I also looked into different security measures that could be taken to ensure user information is handled securely.

- Fabiola Z. Torres Maldonado

**Comment by Cabinetman** [comment]
Researched JavaScript and picked it as the Database back end language.

-Michael Martinez Curbelo

**Comment by gcal77** [comment]
Created tasks for research regarding lazy loading images for faster page load times, as well as implementing basic error pages such as error 404 (Page Not Found) and error 500 (Server Error). Additionally, I spent the week refreshing my knowledge on HTML and learning Javascript.

**Comment by ActuallySeb** [comment]
Set up environment developers will be working in through out the project.

-Sebastian Montalvo Nieves

**Comment by F3rnand0hr** [comment]
Started familiarizing myself with web development features, watched videos, and rewrote some code I made a few months ago when I first had interest in the topic. Completed the task that toggled the visibility of the password when user logs in. Currently finding new ways to implement it efficiently in the sign up feature when it's completed.

**Comment by JoshDG03** [comment]
For this week I had researched what code editor and extensions the DB team was going to be using
